### PR TITLE
Add colorize and palettize to XRImage enhancement_history

### PR DESCRIPTION
**Pull request by djhoese**
I'm 90% sure of the logic on this, but would appreciate a double check. I would like this functionality for my own scripts in my Polar2Grid and Geo2Grid projects. With this the geotiff scale and offset metadata values include the range of the colormap used in palettize or colorize. I can then use that to determine the min/max of the original data so a colorbar can be added (with pydecorate) with the correct limit labels.

 - [ ] Closes #xxxx (remove if there is no corresponding issue, which should only be the case for minor changes)
 - [x] Tests added (for all bug fixes or enhancements)
 - [x] Tests passed (for all non-documentation changes)
 - [x] Passes ``git diff origin/master **/*py | flake8 --diff`` (remove if you did not edit any Python files)
 - [ ] Fully documented (remove if this change should not be visible to users, e.g., if it is an internal clean-up, or if this is part of a larger project that will be documented later)

### CODE REVIEW

File: trollimage/tests/test_image.py [hunk 1]
```diff
@@ -1940,6 +1940,10 @@ def test_colorize(self):
               2.33656886e-01, 2.10634733e-01, 1.88238767e-01]]])
 
         np.testing.assert_allclose(values, expected)
+        assert "enhancement_history" in img.data.attrs
+        assert img.data.attrs["enhancement_history"][-1]["scale"] == 1.0
+        assert img.data.attrs["enhancement_history"][-1]["offset"] == 0.0
```
Comment (pnuu) [inline trollimage/tests/test_image.py:1945]: Should there be a test where the values are different from `1.0` and `0.0`?

**Review by pnuu — APPROVED** [review]
Just one comment, otherwise LGTM.

File: trollimage/tests/test_image.py (hunk 1 above)
Comment (djhoese) [reply trollimage/tests/test_image.py]: I've added some more tests. Let me know what you think.

**Review by pnuu — APPROVED** [review]
LGTM

**Pull request merged**